docs/reference/evaluation.md

```diff
@@ -1,8 +1,56 @@
 # Evaluation Configuration
 
 An [evaluation](#armory.evaluation.Evaluation) completely defines how Armory
-will evaluate a model. It includes the data to be used, the perturbations that
-will be applied to the data, and what metrics will be collected on the model
-outputs.
+will evaluate a model. An evaluation is comprised of chains. Each chain is
+independent, and identifies the dataset input, the perturbations that will be
+applied to the dataset samples, the model, what metrics will be calculated on
+the model outputs, and any exports of samples or predictions.
+
+Chains are defined on an evaluation object using the `add_chain` context.
+
+```python
+from armory.evaluation import Evaluation
+
+evaluation = Evaluation(name="...", description="...", author="...")
+
+with evaluation.add_chain("benign") as chain:
+    chain.use_dataset(...)
+    chain.use_model(...)
+    chain.use_metrics(...)
+    chain.use_exporters(...)
+
+with evaluation.add_chain("attack") as chain:
+    chain.use_dataset(...)
+    chain.use_perturbations(...)  # this chain has input perturbations
+    chain.use_model(...)
+    chain.use_metrics(...)
+    chain.use_exporters(...)
+```
+
+When a component is shared between chains, it can be declared a default for the
+evaluation. If a chain does not specify or override the component inside
+`add_chain`, the default component will be applied to the chain.
+
+```python
+from armory.evaluation import Evaluation
+
+evaluation = Evaluation(name="...", description="...", author="...")
+
+# Common components
+chain.use_dataset(...)
```

Comment:
> it is not clear where chain is defined in this example. Would it be clearer still as `evaluation.use_metrics(…)` etc.?

I'm not sure I understand. The chains in this example are on line 45 and 48.

> This also seems incongruous with experiment_tracking.md#Tracking_Contexts

It's not incompatible with the tracking docs, but deliberately ignored parameter tracking entirely in order to not introduce too many concepts at once (since the tracking docs are quite thorough on that matter already). If you think that this approach is more misleading, I can revise the example (I don't feel too strongly either way).

Reply:
> it is not clear where chain is defined in this example. Would it be clearer still as `evaluation.use_metrics(…)` etc.?

ohhhhh, I see it now. copy-paste error, will fix.

Reply:
> It's not incompatible with the tracking docs, but deliberately ignored parameter tracking entirely...

I take this back. The `evaluation.use_*` methods aren't expected to be called from a tracking context. This is the example shown in the referenced tracking contexts doc:

```python
with evaluation.autotrack() as track_call:
    model = track_call(load_model, name=..., batch_size=...)

with evaluation.autotrack() as track_call:
    dataset = track_call(TheDataset, batch_size=...)

# All chains will receive the dataset's tracked parameters
evaluation.use_dataset(dataset)

with evaluation.add_chain(...) as chain:
    # Only this chain will receive the model's tracked parameters
    chain.use_model(model)
```

So it's just that this example simply doesn't show the _creation_ of _any_ trackable components and is only showing the `evaluation.use_` and `chain.use_` methods that accept the trackables.